RadarSDK/RadarLog.m

```diff
@@ -0,0 +1,75 @@
+//
+//  RadarLog.m
+//  RadarSDK
+//
+//  Copyright © 2021 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarLog.h"
+
+@implementation RadarLog
+
+- (instancetype _Nullable)initWithLevel:(RadarLogLevel)level
+                                message:(NSString *_Nullable)message {
+    self = [super init];
+    if (self) {
+        _level = level;
+        _message = message;
+        _createdAt = [NSDate new];
+    }
+    return self;
+}
+
++ (NSString *)stringForLogLevel:(RadarLogLevel)level {
+    NSString *str;
+    switch (level) {
+        case RadarLogLevelNone:
+            str = @"none";
+            break;
+        case RadarLogLevelError:
+            str = @"error";
+            break;
+        case RadarLogLevelWarning:
+            str = @"warning";
+            break;
+        case RadarLogLevelInfo:
+            str = @"info";
+            break;
+        case RadarLogLevelDebug:
+            str = @"debug";
+            break;
+    }
+    return str;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    dict[@"level"] = [RadarLog stringForLogLevel:self.level];
+    dict[@"message"] = self.message;
+
+    // convert (NSDate)createdAt to ISO8601 string
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"];
+    dateFormatter.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"UTC"];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZ"];
+
+    NSString *createdAtString = [dateFormatter stringFromDate:self.createdAt];
+    dict[@"createdAt"] = createdAtString;
+
+    return dict;
+}
+
++ (NSArray<NSDictionary *> *)arrayForLogs:(NSArray<RadarLog *> *)logs {
+    if (!logs) {
+        return nil;
```

Comment:
nit: should the return value here be `[]` in the empty case?

Reply:
hm ill defer to others, was cargo culting the other `arrayForX`'s which return nil https://github.com/radarlabs/radar-sdk-ios/blob/master/RadarSDK/RadarGeofence.m#L182-L185

in this case it shouldn't be hit either way since higher up, the sync logs call is skipped if no logs to send

Reply:
I thought the same thing, but then I saw that it's the same pattern in the other models.